app/models/user.rb

```diff
@@ -445,6 +446,24 @@ def self.try_to_auto_create_user(login, password)
     user
   end
 
+  def password_changed
+    @password_changed ||= nil
```

Comment:
Unsure what `||= nil` is achieving here, should this just be an `attr_reader :password_changed` ?